apps/server/src/utils/ElasticSearchClient.ts

```diff
@@ -0,0 +1,114 @@
+import type { estypes } from '@elastic/elasticsearch'
+import { Client } from '@elastic/elasticsearch'
+import { readFileSync } from 'fs'
+import type { LocalInstitution } from 'src/shared/contract'
+
+export default class ElasticsearchClient {
+  private readonly client: Client
+  private static _instance: ElasticsearchClient
+
+  private constructor () {
+    this.client = new Client({ node: 'http://localhost:9200' })
+  }
+
+  static async initialize () {
+    const client = this.getInstance().client
+
+    const elasticSearchLoaded = await client.indices.exists({ index: 'institutions' })
+    if (!elasticSearchLoaded) {
+      await this.reIndexElasticSearch()
+    }
+  }
+
+  static getInstance () {
+    if (this._instance) {
+      return this._instance
+    }
+
+    this._instance = new ElasticsearchClient()
+    return this._instance
+  }
+
+  static async reIndexElasticSearch () {
+    const client = this.getInstance().client
+    try {
+      await client.indices.delete({
+        index: 'institutions'
+      })
+    } catch {
+      console.log('Elasticsearch "institutions" index did not exist')
+    }
+    await client.indices.create({ index: 'institutions' })
+    console.log('Elasticsearch indexing institutions')
+    const dataFilePath = 'ucw_institution_mapping.json'
```

Comment:
Can we name file what we want it to be named after MVP 1? Like "defaultInstitutions.json"

Reply:
Yeah, I agree. The filename should probably reflect the date it was generated, or some type of versioning.

Reply:
Yeah I was thinking the same thing about a timestamp. Changing the filename would be easy.

Reply:
I don't think I'm a fan of the timestamp. It adds complexity to loading the file and introduces the potential for people to leave extra copies of their defaults. Everyone should be using versioning, and that will tell them how stale their data is.